api/desecapi/pdns.py

```diff
@@ -191,6 +192,86 @@ def get_rrset_datas(domain):
     ]
 
 
+def update_catalog(zone, delete=False):
+    """
+    Updates the catalog zone information (`settings.CATALOG_ZONE`) for the given zone.
+    """
+    content = _pdns_patch(
+        NSMASTER,
+        "/zones/" + pdns_id(settings.CATALOG_ZONE),
+        {"rrsets": [construct_catalog_rrset(zone=zone, delete=delete)]},
+    )
+    metrics.get("desecapi_pdns_catalog_updated").inc()
+    return content
+
+
+def create_zone_lord(name):
+    name = name.rstrip(".") + "."
```

Comment:
Consider computing this in `CreateDomain.pdns_do()` (which is also pdns-specific) and passing it here and to `create_zone_master`

Reply:
AFAIK, the following HTTP request cannot be used without the final dot on the name. Moving this away would make the API exposed by pdns.py more error-prone.

Reply:
sure, fine with me